loaders/src/glTF/2.0/babylon.glTFLoader.ts

```diff
@@ -209,25 +184,113 @@ module BABYLON.GLTF2 {
         }
 
+        private _setupData(): void {
+            ArrayItem.Assign(this._gltf.accessors);
+            ArrayItem.Assign(this._gltf.animations);
+            ArrayItem.Assign(this._gltf.buffers);
+            ArrayItem.Assign(this._gltf.bufferViews);
+            ArrayItem.Assign(this._gltf.cameras);
+            ArrayItem.Assign(this._gltf.images);
+            ArrayItem.Assign(this._gltf.materials);
+            ArrayItem.Assign(this._gltf.meshes);
+            ArrayItem.Assign(this._gltf.nodes);
+            ArrayItem.Assign(this._gltf.samplers);
+            ArrayItem.Assign(this._gltf.scenes);
+            ArrayItem.Assign(this._gltf.skins);
+            ArrayItem.Assign(this._gltf.textures);
+
+            if (this._gltf.nodes) {
+                const nodeParents: { [index: number]: number } = {};
+                for (const node of this._gltf.nodes) {
+                    if (node.children) {
+                        for (const index of node.children) {
+                            nodeParents[index] = node._index;
+                        }
+                    }
+                }
+
+                const rootNode = this._createRootNode();
+                for (const node of this._gltf.nodes) {
+                    const parentIndex = nodeParents[node._index];
+                    node._parent = parentIndex === undefined ? rootNode : this._gltf.nodes[parentIndex];
+                }
+            }
+        }
+
+        private _createRootNode(): ILoaderNode {
+            this._rootBabylonMesh = new Mesh("__root__", this._babylonScene);
+            const rootNode = { _babylonMesh: this._rootBabylonMesh } as ILoaderNode;
+            switch (this.coordinateSystemMode) {
+                case GLTFLoaderCoordinateSystemMode.AUTO: {
+                    if (!this._babylonScene.useRightHandedSystem) {
+                        rootNode.rotation = [0, 1, 0, 0];
+                        rootNode.scale = [1, 1, -1];
+                        GLTFLoader._LoadTransform(rootNode, this._rootBabylonMesh);
+                    }
+                    break;
+                }
+                case GLTFLoaderCoordinateSystemMode.FORCE_RIGHT_HANDED: {
+                    this._babylonScene.useRightHandedSystem = true;
+                    break;
+                }
+                default: {
+                    throw new Error("Invalid coordinate system mode " + this.coordinateSystemMode);
+                }
+            }
+
+            this.onMeshLoadedObservable.notifyObservers(this._rootBabylonMesh);
+            return rootNode;
+        }
+
+        private _loadNodesAsync(nodes: ILoaderNode[], ): Promise<void> {
+            const promises = new Array<Promise<void>>();
+
+            for (let node of nodes) {
+                promises.push(this._loadNodeAsync("#/nodes/" + node._index, node));
+            }
+
+            promises.push(this._loadAnimationsAsync());
+
+            return Promise.all(promises).then(() => {});

```

Comment:
Do we need the empty lambda could we not return the promise.all directly ?

Reply:
I researched this a bit and didn't find any good answers. If I don't do this, it won't compile since the return value is expected to be Promise<void>. Promise.all returns Promise<T[]>